internal/services/tesla_fleet_api_service.go

```diff
@@ -155,6 +176,66 @@ func (t *teslaFleetAPIService) WakeUpVehicle(ctx context.Context, token, region
 	return nil
 }
 
+func (t *teslaFleetAPIService) RegisterToTelemetryServer(ctx context.Context, token, region, vin string) error {
+	baseURL := fmt.Sprintf(t.Settings.TeslaFleetURL, region)
+	u, err := url.Parse(fmt.Sprintf("%s/api/1/vehicles/fleet_telemetry_config", baseURL))
+	if err != nil {
+		return err
+	}
```

Comment:
Since we do not have control over the baseURL setting we should be able to handle if there is a trailing slash or not this can be done by using URL.JoinPath(). I wrote an example of how a trailing slash breaks this code
https://go.dev/play/p/ChKqJMYcrgt

Reply:
Not sure I understand what you mean by not having control over the baseURL setting

Reply:
The baseURL comes from a setting file that is read in at runtime, so this function has no control over whether or not there is a trailing slash. Using Sprintf like you have now is what caused this bug in identity API https://linear.app/dimo/issue/SI-2524/identity-api-image-creates-a-malformed-url

Reply:
so the base url looks like this `https://fleet-api.prd.%s.vn.cloud.tesla.com` where `%s`  a **region** argument passed from the client. So we still need to use the `fmt.Sprintf`. However, it might also be a good idea to use url.Join